### PR TITLE
Development - add assign labels

### PR DESCRIPTION
**Pull request by yashodgayashan**

### CODE REVIEW

File: backend/services/src/github_services/utilities.bal [hunk 1]
```diff
@@ -212,3 +235,54 @@ function createFormattedCollaborators(json[] collaborators) returns json | error
...
+    }
+
+}
+
+
+
+
+
+
+
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal]: Remove these unwanted new lines.

File: backend/services/src/github_services/utilities.bal [hunk 2]
```diff
@@ -42,11 +45,31 @@ public function checkLabel(string labelName) returns @untainted string[] {
     }
 }
 
-# The `createLabel` function will create a label in a specified git repository.
-#
-# + labelName        - Name of the label.
-# + labelDescription - Description of the label.
-# + return           - Returns a **string[]** which indicates the status.
+
+public function checkLabels(string labelName) returns @untainted string[] {
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal:49]: Whats the purpose of this method? It just returns an array.

File: backend/services/src/github_services/utilities.bal [hunk 3]
```diff
@@ -42,11 +45,31 @@ public function checkLabel(string labelName) returns @untainted string[] {
...
+    return [];
+}
+
+# The `toStringArray` function will convert a json array into string array.
+# 
+# + inputArray - Json array.
+#
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal]: Remove this line.

File: backend/services/src/github_services/utilities.bal [hunk 4]
```diff
@@ -42,11 +45,31 @@ public function checkLabel(string labelName) returns @untainted string[] {
...
+public function toStringArray(json[] inputArray) returns string[] {
+
+    string[] outputArray = [];
+    foreach var item in inputArray {
+        outputArray[outputArray.length()] = item.toString();
+    }
+    return outputArray;
+}
+
+# The `createLabel` Function will create a label in github in particular repositary.
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal]:
```suggestion
# The `createLabel` function will create a label in a specified git repository.
```

File: backend/services/src/github_services/utilities.bal (hunk 2 above)
Comment (yashodgayashan) [reply backend/services/src/github_services/utilities.bal:49]: This method return whether the label is available and the string containing returned status code and the message

File: backend/services/src/github_services/utilities.bal [hunk 5]
```diff
@@ -42,11 +45,31 @@ public function checkLabel(string labelName) returns @untainted string[] {
...
+# 
+# + labelName - The creating label name.
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal]:
```suggestion
# + labelName        - Name of the label.
```

File: backend/services/src/github_services/utilities.bal [hunk 6]
```diff
@@ -42,11 +45,31 @@ public function checkLabel(string labelName) returns @untainted string[] {
...
+# + labelDescription - The description of the label
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal]:
```suggestion
# + labelDescription - Description of the label.
```

File: backend/services/src/github_services/utilities.bal [hunk 7]
```diff
@@ -42,11 +45,31 @@ public function checkLabel(string labelName) returns @untainted string[] {
...
+#
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal:67]: Remove unwanted line here.

File: backend/services/src/github_services/utilities.bal [hunk 8]
```diff
@@ -42,11 +45,31 @@ public function checkLabel(string labelName) returns @untainted string[] {
...
+    }
+    return outputArray;
+}
+
+# The `createLabel` Function will create a label in github in particular repositary.
+# 
+# + labelName - The creating label name.
+# + labelDescription - The description of the label
+# 
+# + return - The `createLabel` function will return **json** to indicate the status.
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal]:
```suggestion
# + return           - Returns a **json** which indicates the status.
```

File: backend/services/src/github_services/utilities.bal [hunk 9]
```diff
@@ -212,3 +235,54 @@ function createFormattedCollaborators(json[] collaborators) returns json | error
     }
     return formattedCollaborators;
 }
+
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal]: Remove extra line here.

File: backend/services/src/github_services/utilities.bal [hunk 10]
```diff
@@ -212,3 +235,54 @@ function createFormattedCollaborators(json[] collaborators) returns json | error
...
+
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal]: Remove extra lines here.

File: backend/services/src/github_services/utilities.bal [hunk 11]
```diff
@@ -212,3 +235,54 @@ function createFormattedCollaborators(json[] collaborators) returns json | error
...
+        if (jsonPayload is json) {
+            return <@untainted>jsonPayload;
+        } else {
+            return error("Error of the jsonPayload.");
+        }
+    } else {
+        return error("Error while retreving labels from the github.");
+    }
+}
+
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal:256]: Remove extra line here.

File: backend/services/src/github_services/utilities.bal (hunk 2 above)
Comment (deshankoswatte) [reply backend/services/src/github_services/utilities.bal:49]: It's not implemented is it?

File: backend/services/src/github_services/utilities.bal [hunk 12]
```diff
@@ -212,3 +233,51 @@ function createFormattedCollaborators(json[] collaborators) returns json | error
...
+    }
+}
+<<<<<<< HEAD
+=======
+
+
+
+
+
+>>>>>>> 654d4c2c345bc53f128d3ea6e124d2fbfa8fcf6f
```
Comment (deshankoswatte) [inline backend/services/src/github_services/utilities.bal]: @yashodgayashan fix this

Comment (yashodgayashan) [reply backend/services/src/github_services/utilities.bal]: Fixed

**Pull request merged**